etcd/server/service.sls

```diff
@@ -9,8 +9,10 @@ etcd_packages:
 {%- if server.get('engine', 'systemd') %}
   - require:
     - file: /etc/default/etcd
+  {% if not grains.get('noservices', False) %}
```

Comment:
This `if` hell makes it very hard to understand for peoplne not familiar with test cases.

I'd prefer to avoid any changes in code for testing. No testing is better than complicated testing IMHO.